leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/CaliforniumEndpointsManager.java

```diff
@@ -172,9 +172,7 @@ public boolean isTrusted(RawPublicKeyIdentity id) {
                     if (this.trustStore != null) {
                         newTrustedCertificatesList.addAll(CertPathUtil.toX509CertificatesList(this.trustStore));
                     }
-                    if (serverInfo.serverCertificate instanceof X509Certificate) {
-                        newTrustedCertificatesList.add((X509Certificate) serverInfo.serverCertificate);
-                    }
+                    newTrustedCertificatesList.add((X509Certificate) serverInfo.serverCertificate);
```

Comment:
I remove the X509 check.
I think if we want to handle this just "not remove it" from the store was not enough.

Reply:
Most likely they all will be X.509 Certificates as really there isn't others in use.